paasta_tools/cli/cmds/validate.py

```diff
@@ -712,15 +712,19 @@ def validate_min_max_instances(service_path):
     return returncode
 
 
-def check_secrets_for_instance(instance_config_dict, soa_dir, service_path, vault_env):
+def check_secrets_for_instance(instance_config_dict, soa_dir, service, vault_env):
     return_value = True
+    # If the service: directive is used, look for the secret there, rather than where the instance config is defined.
+    service_containg_secret = instance_config_dict.get("service", service)
     for env_value in instance_config_dict.get("env", {}).values():
         if is_secret_ref(env_value):
             secret_name = get_secret_name_from_ref(env_value)
             if is_shared_secret(env_value):
                 secret_file_name = f"{soa_dir}/_shared/secrets/{secret_name}.json"
             else:
-                secret_file_name = f"{service_path}/secrets/{secret_name}.json"
```

Comment:
huh, now I'm confused as to how this worked before (and/or why we needed to use `soa_dir` for shared secrets but not for per-service secrets)

Reply:
service_path is both the soa_dir and service name together!

Reply:
I changed input arg to simply service_name

Reply:
ahhh - gotcha!